pkg/operator/ceph/csi/csi.go

```diff
@@ -317,5 +325,16 @@ func (r *ReconcileCSI) setParams(ver *version.Info) error {
 
 	CSIParam.DriverNamePrefix = k8sutil.GetValue(r.opConfig.Parameters, "CSI_DRIVER_NAME_PREFIX", r.opConfig.OperatorNamespace)
 
+	_, err = r.context.ApiExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), "volumegroupsnapshotclasses.groupsnapshot.storage.k8s.io", metav1.GetOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to get volumegroupsnapshotclasses.groupsnapshot.storage.k8s.io CRD")
+	}
+	CSIParam.VolumeGroupSnapshotSupported = err == nil
```

Comment:
Why is it necessary to seek out the CRD for this feature instead of relying on a simple `enabled: true/false`?

Reply:
The VolumeGroupSnapshot feature is currently in Alpha state, and the CRDs may not be available if the feature was not explicitly enabled in the Kubernetes CSI external-snapshotter controller. When a CSI-drivers uses the external-snapshotter sidecar and enables the feature there, it will fail to start because of the missing CRDs. This prevents any snapshot operations from working.

Reply:
I hope the above answers the question of why we have this extra check.